src/Compression/CompressionCodecEncrypted.cpp

```diff
@@ -111,32 +121,18 @@ namespace DB
         return std::string(buffer.data());
     }
 
-    std::string CompressionCodecEncrypted::deriveKey(const std::string_view & master_key)
-    {
-        std::string_view salt(""); // No salt: derive keys in a deterministic manner.
-        std::string_view info("Codec Encrypted('AES-128-GCM-SIV') key generation key");
-        std::array<char, 32> result;
-
-        const int ok = HKDF(reinterpret_cast<uint8_t *>(result.data()), result.size(),
-                            EVP_sha256(),
-                            reinterpret_cast<const uint8_t *>(master_key.data()), master_key.size(),
-                            reinterpret_cast<const uint8_t *>(salt.data()), salt.size(),
-                            reinterpret_cast<const uint8_t *>(info.data()), info.size());
-        if (!ok)
-            throw Exception(lastErrorString(), ErrorCodes::OPENSSL_ERROR);
-
-        return std::string(result.data(), 16);
-    }
-
     void CompressionCodecEncrypted::encrypt(const std::string_view & plaintext, char * ciphertext_and_tag)
     {
-        // Fixed nonce. Yes this is unrecommended, but we have to live
-        // with it.
-        std::string_view nonce("\0\0\0\0\0\0\0\0\0\0\0\0", 12);
+        // randomly generated nonce. Split 12 bytes, because it is too big value and cannot be converted.
+        // Some values are to large for char, so static_cast resolves this problem
+        std::string random_nonce {static_cast<char>(0x3a), 0x3b, static_cast<char>(0xcb), static_cast<char>(0x9d),
+         static_cast<char>(0x7b), 0x43, 0x47, static_cast<char>(0x99), static_cast<char>(0x8c), 0x42, 0x4f, 0x27};
+        std::string_view nonce(random_nonce);
```

Comment:
The same can be done with one line only:

```
std::string_view nonce{"\x3a\x3b...", 16);
```

Reply:
I've thought about it a bit more. A random hard-coded nonce isn't going to add any security anyway because it will be published in open-source code. So probably it's better to go back to zero nonce.